app.go

```diff
@@ -193,9 +198,43 @@ func New(opts ...Option) *App {
 		app.provide(p)
 	}
 	app.provide(func() Lifecycle { return app.lifecycle })
+
+	if app.err != nil {
+		app.logger.Printf("Error after options were applied: %v", app.err)
+		return app
+	}
+
+	if err := app.executeInvokes(); err != nil {
+		app.err = err
```

Comment:
I really dont want to return an err from New if possible.

I'm OK with the idea of `fx.TryNew` or `app.Error`.

@glibsm @akshayjshah @abhinav can you weigh in here?